private/buf/cmd/buf/command/registry/registrylogin/registrylogin.go

```diff
@@ -165,6 +169,17 @@ func inner(
 		}
 		token = string(data)
 	} else {
+		// Check if we already have a valid token for this remote and username.
+		if machine, err := netrc.GetMachineForName(container, remote); machine != nil && err == nil {
```

Comment:
Note that this is not actually getting a token for the given remote/username pair, by definition the API is just getting machine based on remote. Then, in `isValidToken`, I error if the username is different. It's common to want to login with a different user, and the general pattern I'd expect to do that is to just call `buf registry login` again. With this PR, I'd now get an error with that (and that error wouldn't tell me to first logout, although that's fixable).

Reply:
On a different username the error would be raised as not matching and continue the normal flow of prompting the user for their token. No error would be raised. It's behaviour is meant to be analogous to a website style "already logged in" no need to re-enter credentials.